### PR TITLE
Feature/payload generator

### PR DESCRIPTION
**Pull request by KevponSAP**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn install` runs for the whole project and, if you touched any code in the respective service, submission and distribution service can be run with `spring-boot:run`

## Description

- Added a submission payload generator
- Added integration tests with multiple different protobuf payloads

### CODE REVIEW

**Review by emmetsap — DISMISSED** [review]

File: services/submission/src/test/java/app/coronawarn/server/services/submission/SubmissionPayloadGenerator.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  public static void main(String[] args) throws IOException {
+    int numberOfKeys = 10;
+    int transmissionRiskLevel = 6;
+    int rollingPeriod = 144; // 24*60/10
+    ReportType reportType = ReportType.CONFIRMED_CLINICAL_DIAGNOSIS;
+    ByteString requestPadding = ByteString.copyFrom(new byte[100]);
+    final List<String> visitedCountries = List.of("DE", "FR");
+    String originCountry = "DE";
+    boolean consentToFederation = true;
+    int daysSinceOnsetOfSymptoms = 0;
```
Comment (mibrasap) [inline services/submission/src/test/java/app/coronawarn/server/services/submission/SubmissionPayloadGenerator.java:31]: Would propose to extract the variables as constants on class level for better readability

File: services/submission/src/test/java/app/coronawarn/server/services/submission/SubmissionPayloadGenerator.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    LocalDateTime now = LocalDateTime.now();
+    LocalDateTime todayMidnight = LocalDateTime
+        .of(now.getYear(), now.getMonth(), now.getDayOfMonth(), 0, 0);
```
Comment (mibrasap) [inline services/submission/src/test/java/app/coronawarn/server/services/submission/SubmissionPayloadGenerator.java:35]: Did you check if LocalDate atStartOfDay() is working here as well?

**Review by mibrasap — CHANGES_REQUESTED** [review]

File: services/submission/src/test/java/app/coronawarn/server/services/submission/SubmissionPayloadGenerator.java (hunk 1 above)
Comment (hilmarf) [reply services/submission/src/test/java/app/coronawarn/server/services/submission/SubmissionPayloadGenerator.java:31]: or even better take some of the arguments as input parameter from `String[] args`

**Review by hilmarf — APPROVED** [review]

File: services/submission/src/test/java/app/coronawarn/server/services/submission/SubmissionPayloadGenerator.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+  private static final int numberOfKeys = 10;
+  private static final int transmissionRiskLevel = 6;
+  private static final int rollingPeriod = 144; // 24*60/10
+  private static final ReportType reportType = ReportType.CONFIRMED_CLINICAL_DIAGNOSIS;
+  private static final ByteString requestPadding = ByteString.copyFrom(new byte[100]);
+  private static final List<String> visitedCountries = List.of("DE", "FR");
+  private static final String originCountry = "DE";
+  private static final boolean consentToFederation = true;
+  private static final int daysSinceOnsetOfSymptoms = 0;
```
Comment (mibrasap) [inline services/submission/src/test/java/app/coronawarn/server/services/submission/SubmissionPayloadGenerator.java:27]: By convention, constants shall be uppercase with underscores, as it is the case for path to the payload below

Comment (KevponSAP) [reply services/submission/src/test/java/app/coronawarn/server/services/submission/SubmissionPayloadGenerator.java]: done

**Review by mibrasap — APPROVED** [review]

**Review by emmetsap — APPROVED** [review]

**Pull request merged**